app.rb

```diff
@@ -0,0 +1,108 @@
+require './person'
+require './book'
+require './student'
+require './teacher'
+require './rental'
+require './classroom'
+require './nameable'
+
+class App
+  def initialize
+    @book = []
+    @people = []
+    @rental = []
+  end
+
+  def list_all_books
+    if @book.empty?
+      puts 'No books available'
+    else
+      @book.each { |book| puts Title: book.title, Author: book.author }
+    end
+  end
+
+  def list_all_people
+    if @people.empty?
+      puts 'No people available'
+    else
+      @people.each { |person| puts Name: person.name, ID: person.id, Age: person.age }
+    end
+  end
+
+  def create_person
+    puts 'Do you want to  create a student (1) or a teacher(2)?'
+    res = gets.chomp.to_i
+
+    if res == 1
+      create_student
+    elsif res == 2
+      create_teacher
+    end
```

Comment:
[OPTIONAL]

- I recommend adding the option when a user provides an invalid parameter.

```
if res != "1" && res != "2"
      puts "Invalid option"
      return
    end
```